examples/clone/src/main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <KDIS/DataTypes/EntityIdentifier.hpp>
+#include <KDIS/PDU/Entity_Info_Interaction/Entity_State_PDU.hpp>
+#include <KDIS/PDU/Header.hpp>
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <type_traits>
+
+static void print(const std::string &label, const KDIS::PDU::Header &pdu);
+
+int main() {
+  // Header
+  KDIS::PDU::Header header;
+  header.SetExerciseID(1);                       // Update exercise id
+  auto header_clone = KDIS::PDU::clone(header);  // Clone
+  header_clone->SetExerciseID(2);                // Update exercise id in clone
+  assert(header.GetExerciseID() != header_clone->GetExerciseID());
```

Comment:
We could probably turn this into a test.

Reply:
Yeah, good idea. Can you try to add something?

Reply:
Yeah sure

Reply:
Thanks 🙏

Reply:
Im getting errors trying to compile the tests

```
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: The command "setlocal
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: "C:\Program Files\CMake\bin\cmake.exe" -D TEST_TARGET=KDIS_tests -D TEST_EXECUTABLE=C:/Users/karl/Documents/KDIS/kdis-github/kdis/tests/Debug/KDIS_tests.exe -D TEST_EXECUTOR= -D TEST_WORKING_DIR=C:/Users/karl/Documents/KDIS/kdis-github/kdis/tests -D TEST_EXTRA_ARGS= -D TEST_PROPERTIES= -D TEST_PREFIX= -D TEST_SUFFIX= -D TEST_FILTER= -D NO_PRETTY_TYPES=FALSE -D NO_PRETTY_VALUES=FALSE -D TEST_LIST=KDIS_tests_TESTS -D CTEST_FILE=C:/Users/karl/Documents/KDIS/kdis-github/kdis/tests/KDIS_tests[1]_tests.cmake -D TEST_DISCOVERY_TIMEOUT=5 -D TEST_XML_OUTPUT_DIR= -P "C:/Program Files/CMake/share/cmake-3.28/Modules/GoogleTestAddTests.cmake"
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: if %errorlevel% neq 0 goto :cmEnd
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: :cmEnd
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: :cmErrorLevel
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: exit /b %1
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: :cmDone
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: if %errorlevel% neq 0 goto :VCEnd
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(159,5): error MSB3073: :VCEnd" exited with code 1.
```

Are they working for you?

Reply:
When you prepare cmake you have to set `CNAKE_BUILD_TESTS=TRUE`. See README.md :)

Reply:
Oh thats strange. Im using the GUI which only has KDIS_BUILD_TESTS

![image](https://github.com/karljj1/kdis/assets/13064524/4b3a575c-1abc-418d-bc82-185962936593)

Reply:
Strange, check the CI (and copy paste) since it works there :/

Reply:
I switched to compiling against the lib instead of dll and it worked :)

Reply:
I added a test for DIS 6, Its pretty basic but better than nothing.

Reply:
Awesome 😎

Reply:
I need to paramatize the other tests

Reply:
Yeah, in only a few lines the tests are easily understandable